app/controllers/attribute.controller.js

```diff
@@ -1,19 +1,15 @@
-var request = require('request');
-var config = require('../config/config');
 var attributeDao = require('../dao/attribute.dao');
 var q = require('q');
 
-
 // TODO: Make the validation more covering
 function validateAttribute(attribute) {
-    var deferred = q.defer();
-    if (attribute && attribute.name) {
-        deferred.resolve(attribute);
-    } else {
-        defered.reject(new Error('Not a valid attribute object!'));
-    }
-    return deferred.promise;
-};
+    q.promise(function(resolve, reject) {
```

Comment:
This change, i.e. switching to q.promise looks pretty great. However, I would have preferred it in a separate pull request as it's a very isolated change